Depends on #108: Add custom Gettext backend

arturz opened the pull request.

vi0dine committed:
chore: :bookmark: bump version to 0.4.1 (#88)

jk-lamb committed:
Fix Pagination UI

- Rewrite the functional component
- Add guards (nonnegative page number and total pages)
- Add "surrounding_pages_number" to set number of the pages to the right and left visible in the UI.
- Always show current page

jk-lamb committed:
Fix redirect to Dashboard

When hit http://localhost:4000/kanta/  - proper redirect to the Dashboard

jk-lamb committed:
Change translation progress calculation to use floor

jk-lamb committed:
Merge pull request #112 from curiosum-dev/99-round-down-completion-percentage

Change translation progress calculation to use floor

jk-lamb committed:
Remove commented Router part.

jk-lamb committed:
Merge pull request #110 from curiosum-dev/104-develop-fix-pagination-ui

Fix Pagination UI

jk-lamb committed:
Remove Redirect module since its unused.

jk-lamb committed:
Merge pull request #111 from curiosum-dev/110-redirect-issue-for-kanta

Fix redirect to Dashboard

jk-lamb committed:
Merge pull request #113 from curiosum-dev/develop

Develop

jk-lamb committed:
Gettext 0.26.0 compatibility

- Add Gettext.Backend implementation (Gettext 0.26.2) for Kanta.
- Kanta Cached DB
- Add Fallback Gettext backend for PO file checks
- Gettext.Recompiler util module to support one time recompilation after mix gettext.extract

Tests:
- Add migration(v4) for test repo
- Re-implemented DataCase helper (manual checkout)
- Add tests for the Backend and Adapter

arturz commented on lib/kanta/utils/module_folder.ex:
```diff
@@ -0,0 +1,46 @@
+  defp module_to_string(module) when is_binary(module), do: module
+
+  defp remove_elixir_prefix("Elixir." <> rest), do: rest
+  defp remove_elixir_prefix(other), do: other
+
+  defp replace_invalid_chars(name, replacement) do
+    # Replace characters that are invalid in folder names
+    name
+    |> String.replace(~r/[^\w\-\.]/, replacement)
+    # Collapse multiple replacements
+    |> String.replace(~r/#{replacement}+/, replacement)
+  end
```
Add `|> String.trim_trailing(".")` to comply with Windows folder naming rules.

```elixir
  defp replace_invalid_chars(name, replacement) do
    # Replace characters that are invalid in folder names
    name
    |> String.replace(~r/[^\w\-\.]/, replacement)
    # Collapse multiple replacements
    |> String.replace(~r/#{replacement}+/, replacement)
    # Comply with Windows naming rules
    |> String.trim_trailing(".")
  end
```

jk-lamb replied:
Roger!

arturz commented on lib/kanta/utils/module_folder.ex:
```diff
@@ -0,0 +1,46 @@
+  """
+  def safe_folder_name(module, opts \\ []) when is_atom(module) or is_binary(module) do
+    lowercase? = Keyword.get(opts, :lowercase, true)
+    replacement = Keyword.get(opts, :replace_with, "_")
```
Since we do not pass `opts` in any of this function's calls, is it necessary?

jk-lamb replied:
Ok.

arturz commented on lib/kanta/backend.ex:
```diff
@@ -0,0 +1,127 @@
+
+      # When `mix gettext extract` create POT/PO files based on this backend usage (ex. getext(...) call) across the application codebase.
+      if Gettext.Extractor.extracting?() do
+        Logger.debug("#{__MODULE__} OFF")
```
This log may be misleading for users. Additionally, I think Kanta shouldn't log any translation events by default, as it would flood the terminal with logs. We could add support for debug logs, similar to how it's configured in [Oban](https://github.com/oban-bg/oban/blob/c26a1f487d536b0051ee63b5d030318753a74f2a/lib/oban.ex#L317) and [Ecto](https://github.com/elixir-ecto/ecto/blob/a1bbc0c66470caa04e16bd7fb82372c106bbcda5/lib/ecto/repo.ex#L42), by passing the `:log` level via configuration. However, that’s likely out of scope for this PR.

jk-lamb replied:
Very good point!

arturz commented on priv/repo/migrations/20250327123641_update_kanta_migrations.exs:
```diff
@@ -0,0 +1,11 @@
+defmodule Kanta.Test.Repo.Migrations.UpdateKantaMigrations do
+  use Ecto.Migration
+
+  def up do
+     Kanta.Migration.up(version: 4)
+   end
+
+   def down do
+     Kanta.Migration.down(version: 4)
+   end
+end
```
This file is likely a mistake, as it should be created in an application using Kanta, not in Kanta itself.

jk-lamb replied:
As discussed with @arturz , this is one necessary to run tests for Kanta and create kanta-test db in order to test the DB adapter.

jk-lamb committed:
Fix: non async tests for DB-related operation.

jk-lamb committed:
Minor fixes

- Remove Logger.debug(...)
- add removing trailing "." to safe_folder_name (for Windows)

szsoppa committed:
Updates company details in README.md

szsoppa committed:
Fixes link to License in README.md

vincentvanbush committed:
Add contributing rules, code of conduct and GH templates

vincentvanbush committed:
Update readme

vincentvanbush committed:
Update conduct file

vincentvanbush committed:
Bump deps, set Phoenix to 1.7+ [#122]

vincentvanbush committed:
Conditional config for Phoenix 1.7/1.8 [#122]

vincentvanbush committed:
Fix Tailwind styles in compiled assets [#122]

vincentvanbush committed:
Update changelog

vincentvanbush committed:
Update README [#122]

vincentvanbush committed:
Change method of checking phoenix version [#122]

vincentvanbush committed:
Merge pull request #123 from curiosum-dev/122-feature-phoenix-18-compatibility

Phoenix 1.8 compatibility [#122]

vincentvanbush committed:
chore: :bookmark: Bump version to 0.4.2

- Fix Dialyzer errors related to Phoenix version detection
- Replace Mix.Dep.Lock.read/0 with Application.spec/2 for better static analysis compatibility
- Add .dialyzer_ignore.exs to handle Phoenix dependency warnings
- Update README to reflect Phoenix 1.7.x and 1.8.x compatibility

jk-lamb committed:
Merge remote-tracking branch 'origin/main' into develop

jk-lamb committed:
Update README.md

jk-lamb committed:
Merge branch 'develop' into 108-base-on-gettext-026

jk-lamb committed:
Update mix.exs: Relax gettext dependency constraint (0.26.*)

jk-lamb committed:
Update README.md

jk-lamb committed:
Update CHANGELOG.md

jk-lamb committed:
Add missing @moduledoc annotations

The changes add missing @moduledoc attributes to various modules,
providing clear documentation about their purpose.

jk-lamb committed:
Update README.md

jk-lamb committed:
Add migration step to GitHub Actions database setup

mix ecto.drop && mix ecto.create && mix ecto.migrate

jk-lamb committed:
Address PR review feedback

- Add breaking change notice to CHANGELOG.md explaining Gettext 0.26.0 migration requirements
- Fix apply_bindings return type from :not_found to {:error, :not_found} to match codebase conventions
- Rename KantaCachedDB to CachedDB module (remove redundant Kanta prefix)
- Update all references to use new module name

jk-lamb committed:
Address PR review feedback

- Add breaking change notice to CHANGELOG.md explaining Gettext 0.26.0
migration requirements - Fix apply_bindings return type from :not_found
to {:error, :not_found} to match codebase conventions - Rename
KantaCachedDB to CachedDB module (remove redundant Kanta prefix) -
Update all references to use new module name

jk-lamb committed:
Move recompilation flags to _build directory

Relocates GettextRecompiler flag files from priv/kanta/ to environment-specific build directories using Mix.Project.build_path(). This prevents confusion about whether temporary build artifacts should be committed to version control, as _build/ is already in .gitignore.

The flag files now reside in:
- _build/{env}/kanta_recompile/.gettext_recompiled
- _build/{env}/kanta_recompile/.fallback_recompiled

jk-lamb committed:
Merge branch '108-base-on-gettext-026' of https://github.com/curiosum-dev/kanta into 108-base-on-gettext-026

# Conflicts:
#	lib/kanta/backend.ex

jk-lamb committed:
Rename cached_db test file to match module name

jk-lamb committed:
Fix line break in cached DB test
